### PR TITLE
3321 device settings

### PR DESCRIPTION
**Pull request by lorenjohnson**

### CODE REVIEW

**Comment by danekszy** [comment]
Is this ready for review? If so, can we resolve conflicts first?

**Comment by lorenjohnson** [comment]
> Is this ready for review? If so, can we resolve conflicts first?

No, and you'd know it was if you were tagged for review :)

File: src/pages/MainMenu/MainMenu.js [hunk 1]
```diff
@@ -12,5 +12,6 @@ export function Menu () {
     <li><Link to='/'>Dashboard</Link></li>
     <li><Link to='/browse-happs'>Browse Happs</Link></li>
     <li><Link to='/pricing'>Manage Pricing</Link></li>
+    <li><Link to='settings'>Settings</Link></li>
```
Comment (robbiecarlton) [inline src/pages/MainMenu/MainMenu.js:15]: add leading slash

File: src/pages/ManagePricing/ManagePricing.js [hunk 2]
```diff
@@ -69,7 +69,7 @@ export default function ManagePricing ({ history: { push } }) {
 
     <div styleName='price-input-wrapper'>
       <label styleName='price-input-label'>
-        Holofuel per unit
+        HoloFuel per unit
```
Comment (robbiecarlton) [inline src/pages/ManagePricing/ManagePricing.js]: 👍

File: src/pages/Settings/Settings.js [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+import useForm from 'react-hook-form'
+import * as yup from 'yup'
+import Button from 'components/Button'
+
+import './Settings.module.css'
+
+export const PORT_NUMBER_REGEX = /^\d{4,6}$/
+export const EMAIL_REGEX = /^\S+@\S+$/i
+
+const SettingsValidationSchema = yup.object().shape({
```
Comment (robbiecarlton) [inline src/pages/Settings/Settings.js:12]: 👍

File: src/pages/Settings/Settings.js [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+})
+
+export function Settings ({
+  settings,
+  updateSettings,
+  factoryReset,
+  toggleSshAccess,
+  history: { push }
+}) {
+  // if (props.loading) return <h4>Loading Settings</h4>
```
Comment (robbiecarlton) [inline src/pages/Settings/Settings.js]: remove comment

Comment (robbiecarlton) [reply src/pages/Settings/Settings.js]: or add loading state ;)

File: src/pages/PrimaryLayout/PrimaryLayout.module.css [hunk 5]
```diff
@@ -1,6 +1,7 @@
 .primary-layout {
   display: flex;
   word-break: break-word;
+  justify-content: center;
```
Comment (robbiecarlton) [inline src/pages/PrimaryLayout/PrimaryLayout.module.css]: This breaks the padding on the settings page. But I can address it as part of the layout task.

File: src/pages/Settings/Settings.js [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+  const goToMenu = () => push('/menu')
+  const { register, handleSubmit, errors } = useForm({
+    defaultValues: settings,
+    validationSchema: SettingsValidationSchema
+  })
+
+  const [sshAccessVal, setSshAccess] = useState(false)
+
+  const handleViewTos = (e) => {
+    e.preventDefault()
```
Comment (robbiecarlton) [inline src/pages/Settings/Settings.js]: I haven't been doing this. What does it achieve?

File: src/pages/Settings/Settings.js [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+
+  const handleToggleSshAccess = (e) => {
+    e.preventDefault()
+    setSshAccess(e.target.checked)
+    toggleSshAccess()
+  }
+
+  const onSubmit = values => {
+    // e.preventDefault()
+    console.log('!!! New settings:', values, errors)
```
Comment (robbiecarlton) [inline src/pages/Settings/Settings.js]: remove comment and console

File: src/pages/Settings/Settings.js [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+    updateSettings(settings)
+  }
+
+  console.log('!!! errors: ', errors)
```
Comment (robbiecarlton) [inline src/pages/Settings/Settings.js]: console

File: src/pages/Settings/Settings.test.js [hunk 9]
```diff
@@ -0,0 +1,7 @@
+// import React from 'react'
+// import Settings from './Settings'
+// import { render } from '@testing-library/react'
+
+describe('Settings', () => {
+  it("isn't tested", () => {})
```
Comment (robbiecarlton) [inline src/pages/Settings/Settings.test.js:6]: what's the plan for testing this? Just leave it til we know what the real settings will be?

File: src/pages/Tos/Tos.test.js [hunk 10]
```diff
@@ -0,0 +1,7 @@
+// import React from 'react'
+// import Settings from './Settings'
+// import { render } from '@testing-library/react'
+
+describe('Settings', () => {
```
Comment (robbiecarlton) [inline src/pages/Tos/Tos.test.js]: not that it matters, but should be 'Tos'

**Review by robbiecarlton — CHANGES_REQUESTED** [review]
Some cleanup, and a couple of questions.

One other change: EnvoyInterface is getting renamed to envoyInterface. `git mv envoyInterface EnvoyInterface` will fix it.

File: src/pages/Settings/Settings.js (hunk 6 above)
Comment (lorenjohnson) [reply src/pages/Settings/Settings.js]: Keeps a form from HTML submitting (page reloading). May or may not be necessary anymore, carry over from Lisa code.

Comment (robbiecarlton) [reply src/pages/Settings/Settings.js]: 👍

File: src/pages/Settings/Settings.js (hunk 4 above)
Comment (lorenjohnson) [reply src/pages/Settings/Settings.js]: removing, this and a few of these things were leftovers from Lisa's work. We will not introduce loading state until we have data.

File: src/pages/PrimaryLayout/PrimaryLayout.module.css (hunk 5 above)
Comment (lorenjohnson) [reply src/pages/PrimaryLayout/PrimaryLayout.module.css]: I will remove it here. It was leftovers again and I don't want to bring it in.

File: src/pages/Settings/Settings.js (hunk 8 above)
Comment (lorenjohnson) [reply src/pages/Settings/Settings.js]: 👍 Leaving this on purpose as for Ugly UI sans data + new forms hook/API it's good to stay aware of it. I'll add a comment to this effect. Don't want to decide right now where we put these errors and will make sure C is including error states so we'll have a target.

File: src/pages/Settings/Settings.test.js (hunk 9 above)
Comment (lorenjohnson) [reply src/pages/Settings/Settings.test.js:6]: I'll put some testing in, just hadn't gotten to it. I was being bad.

File: src/pages/Tos/Tos.test.js (hunk 10 above)
Comment (lorenjohnson) [reply src/pages/Tos/Tos.test.js]: Got it and I'll introduce a baseline test here as well.

File: src/pages/Settings/Settings.js [hunk 11]
```diff
@@ -0,0 +1,153 @@
...
+          label='HC Network'
+          name='hcNetworkPort'
+          register={register} />
+
+        <SettingsFormInput
+          label='Hosting'
+          name='hostingPort'
+          register={register} />
+
+        <Button type='submit' name='update-settings' value='Submit'>Update</Button>
```
Comment (robbiecarlton) [inline src/pages/Settings/Settings.js:101]: This is our Button react component, not a native HTML button, which currently doesn't take `type`, `submit` or `value` props. I'm modifying this component to take `type` in my branch.

**Pull request merged**